rust/chains/hyperlane-ethereum/src/validator_announce.rs

```diff
@@ -132,21 +133,24 @@ where
         Ok(storage_locations)
     }
 
-    async fn announce_tokens_needed(
-        &self,
-        announcement: SignedType<Announcement>,
-    ) -> ChainResult<U256> {
+    #[instrument(ret, skip(self))]
+    async fn announce_tokens_needed(&self, announcement: SignedType<Announcement>) -> Option<U256> {
         let validator = announcement.value.validator;
-        let contract_call = self.announce_contract_call(announcement, None).await?;
-        if let Ok(balance) = self.provider.get_balance(validator, None).await {
-            if let Some(cost) = contract_call.tx.max_cost() {
-                Ok(cost.saturating_sub(balance))
-            } else {
-                Err(ProviderError::CustomError("Unable to get announce max cost".into()).into())
-            }
-        } else {
-            Err(ProviderError::CustomError("Unable to query balance".into()).into())
-        }
+        let Ok(contract_call) = self
+            .announce_contract_call(announcement, None)
+            .await else {
```

Comment:
So I know there is no official rustfmt standard on this just yet, but I would prefer if the `else` is on the next line when the first line gets split. At least until rust officially declares something.

Much cleaner than it was though!